inbm-lib/inbm_common_lib/utility.py

```diff
@@ -124,7 +124,7 @@ def remove_file(path: Union[str, Path]) -> bool:
         return False
 
 
-def create_file_with_contents(path: Union[str, Path], contents: List[str]) -> None:
+def create_file_with_contents(path: Union[str, Path], contents: list[str]) -> None:
```

Comment:
**Suggestion:** Use 'pathlib.Path' objects consistently for file system paths to make the code more idiomatic and readable. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
canonical_path = get_canonical_representation_of_path(path)
```